javagen/src/main/java/com/azure/autorest/template/example/ProtocolTestWriter.java

```diff
@@ -106,7 +101,7 @@ public ProtocolTestWriter(TestContext testContext) {
                 JavaIfBlock codeBlock = methodBlock.ifBlock("getTestMode() == TestMode.PLAYBACK", ifBlock -> {
                     if (isTokenCredential) {
                         ifBlock.line(String.format("%1$s.httpClient(interceptorManager.getPlaybackClient())", builderVarName));
-                        ifBlock.line(".credential(request -> Mono.just(new AccessToken(\"this_is_a_token\", OffsetDateTime.MAX)));");
+                        ifBlock.line(".credential(new MockTokenCredential());");
```

Comment:
this change is not related

just feel better to use the new `MockTokenCredential` in azure-core-test